### PR TITLE
Run analyzer on the elm repo in the CI

### LINKED ISSUE #65: Run the rules on all of the exercises in `exercism/elm` in the CI
We should run all of the rules on the example/examplar code in `exercism/elm` in the CI to make sure that we are up to our own standards.

There is already something like that running with the test-runner, for each PR on the test-runner, the CI pulls `exercism/elm` and checks that everything passes, and also the other way around, for PRs on  `exercism/elm` the CI pulls the test runner and checks. Look at those CI processes for inspiration and do the same for the analyzer.

### PR DESCRIPTION
**Pull request by jiegillet**
Closes #65 

This can't work in the CI until https://github.com/exercism/elm/pull/633 is merged, because it uses a script in there.
Edit: the PR is merged, and the script seems to work fine.

### CODE REVIEW

**Review by ceddlyburge — APPROVED** [review]

**Comment by jiegillet** [comment]
@exercism/maintainers-admin can I get an approval/merge please? 💛

File: .github/workflows/elm_tests.yml
Comment (ErikSchierboom) [inline .github/workflows/elm_tests.yml:139]: I would suggest building the Docker image here in CI, which will allow it to be cached between run. 
See https://github.com/exercism/generic-representer/blob/main/.github/workflows/ci.yml#L19-L32 for an example (note: the image name must be the same here and in the analyze-all-exercises-in-docker.sh script)

**Review by ErikSchierboom — APPROVED** [review]

File: .github/workflows/elm_tests.yml
Comment (jiegillet) [reply .github/workflows/elm_tests.yml:139]: Thank you, I'll take a look.
But does it really make sense to cache it when the source code will most likely change in most PRs?

Comment (ErikSchierboom) [reply .github/workflows/elm_tests.yml:139]: Well, you'd still get to use the Docker cache for these lines: https://github.com/exercism/elm-analyzer/blob/main/Dockerfile#L1-L28

So I think it would still be useful

Comment (jiegillet) [reply .github/workflows/elm_tests.yml:154]: @ErikSchierboom I added the caching, can you have a look?

File: bin/analyze-all-exercises-in-docker.sh
Comment (ErikSchierboom) [inline bin/analyze-all-exercises-in-docker.sh:12]: This should match the image name in the GHA workflow:

```suggestion
docker build --rm -t exercism/elm-analyzer .
```

File: bin/analyze-all-exercises-in-docker.sh
Comment (ErikSchierboom) [inline bin/analyze-all-exercises-in-docker.sh:23]: This should match the image name in the GHA workflow:

```suggestion
    exercism/elm-analyzer \
```

File: .github/workflows/elm_tests.yml
Comment (ErikSchierboom) [reply .github/workflows/elm_tests.yml:154]: LGTM!

**Pull request merged**
